proposals/0000-linear-types.rst

```diff
@@ -448,6 +394,92 @@ Then ``xi`` has multiplicity annotation ``p*qi``. For instance
   bar (x,y) = … -- Since (,) :: a ->. b ->. (a,b), x and y have
                 -- multiplicity p
 
+.. _Patterns
+
+Deep patterns & multiple equations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Type-checking deep patterns naturally extends the simple patterns
+above. For instance in
+
+::
+
+  f :: Maybe (a, b) ->. …
+  f (Just (x,y)) = …
+
+since the type annotation on the first argument is linear, the outer
+pattern is type-checked as a ``case_1``:
+
+::
+
+  f mxy = case_1 mxy of
+    Just xy -> …
+
+Therefore, the generated intermediate variable ``xy`` has multiplicity
+1, therefore, the inner pattern is elaborated as a ``case_1`` (that is
+the same multiplicity as the intermediate variable).
+
+::
+
+  f mxy = case_1 mxy of
+    Just xy -> case_1 xy of
+      (x, y) -> …
+
+Unresolved questions
+++++++++++++++++++++
+
```

Comment:
I am not sure at all what the point is here. If there is an issue with the translation to Core, then the translated code need to be shown. Furthermore, I am not convinced that there is any significant issue. (I wrote an email on Nov. 21 2017 about a simple solution).